### PR TITLE
Eslint rule updates

### PR DESCRIPTION
**Pull request by toasted-nutbread**
This change adds a few more eslint rules to prevent some simple cases of stylistic deviation. A few places in the codebase were adjusted to accommodate these rules, but for the most part the additional rules already match like 99% of the codebase.

There are several `@typescript-eslint/*` rules declared that have the value "off". These rules use type analysis and when they are on, they require something like ~900 places in the codebase to be evaluated. This can be done independently from this PR.

### CODE REVIEW

File: ext/js/pages/settings/profile-controller.js [hunk 1]
```diff
@@ -549,7 +549,8 @@ export class ProfileController {
             Number.isFinite(intValue) &&
             intValue >= 0 &&
             intValue < this.profileCount ?
-            intValue : null
+            intValue :
```
Comment (djahandarie) [inline ext/js/pages/settings/profile-controller.js:552]: Personally, forcing these to line break seems a bit overkill to me

File: ext/js/templates/sandbox/anki-template-renderer.js [hunk 2]
```diff
@@ -531,6 +534,7 @@ export class AnkiTemplateRenderer {
     _concat(args) {
         let result = '';
         for (let i = 0, ii = args.length; i < ii; ++i) {
+            // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
```
Comment (djahandarie) [inline ext/js/templates/sandbox/anki-template-renderer.js:537]: I'm surprised this is needed, isn't everything a string here?

File: ext/js/data/options-util.js [hunk 3]
```diff
@@ -495,6 +496,7 @@ export class OptionsUtil {
      * @returns {import('options-util').UpdateFunction[]}
      */
     _getVersionUpdates(targetVersion) {
+        /* eslint-disable @typescript-eslint/unbound-method */
```
Comment (djahandarie) [inline ext/js/data/options-util.js:499]: Huh, why is this needed? Are we intentionally using an unbound `this` here? If so it'd probably be better to take in an arg or something

**Review by djahandarie — APPROVED** [review]
LGTM just have a few questions

File: ext/js/data/options-util.js (hunk 3 above)
Comment (toasted-nutbread) [reply ext/js/data/options-util.js:499]: None of these are bound because they are bound later on, as it is used purely internally.

https://github.com/themoeway/yomitan/blob/805cf9cb3ea744a6d7c0a5da27fc9ef4e6f08626/ext/js/data/options-util.js#L485

File: ext/js/templates/sandbox/anki-template-renderer.js (hunk 2 above)
Comment (toasted-nutbread) [reply ext/js/templates/sandbox/anki-template-renderer.js:537]: It's needed because `args` is `unknown[]`. I would have made it so that this disable wasn't necessary, but there's no way to specifically allow `unknown` stringification without disabling the rule, and I think this was the only place that had an issue, so I just suppressed this location.

File: ext/js/pages/settings/profile-controller.js (hunk 1 above)
Comment (toasted-nutbread) [reply ext/js/pages/settings/profile-controller.js:552]: The reason I made this change was to just enforce consistency, but I can remove it if you feel strongly about it. There are only 9 situations in the codebase where a conflicting pattern was used, and about ~40 places where a 3-line ternary was used.

Here is an example of something that should definitely never be a single line:

https://github.com/themoeway/yomitan/blob/805cf9cb3ea744a6d7c0a5da27fc9ef4e6f08626/ext/js/app/popup.js#L807-L811

So the intent was to just try to make sure things like that cannot be made into a single line.
